roles/v2/mining-proxy/src/lib/upstream_mining.rs

```diff
@@ -421,6 +602,37 @@ impl UpstreamMiningNode {
         setup_connection.try_into().unwrap()
     }
 
+    pub fn open_standard_channel_down(
+        &mut self,
+        request_id: u32,
+        downstream_hash_rate: f32,
+        id_header_only: bool,
+    ) -> Vec<Mining<'static>> {
+        match self.channel_kind {
+            ChannelKind::Group => panic!(),
+            ChannelKind::Extended => {
+                //TODO: REVIEW id field set to 0
```

Comment:
Is this TODO for you?

Reply:
Yes,  it was for me but I'm not sure about it. Maybe @Fi3 can you help ? thanks

Reply:
`open_standard_channel_down` is called by `crate::downstream_mining::DownstreamMiningNode::handle_open_standard_mining_channel`  the `DownstreamMiningNode` struct has a field named `status` this field can be either `Initializing`  `Paired(CommonDownstreamData)` `ChannelOpened(Channel)`.
`handle_open_standard_mining_channel` set this field to `ChannelOpened(Channel)` based on the response received by `open_standard_channel_down` if upstream connection is Group we get this value from upstream. 
If upstream connection is Extended or ExtendedWithNegotiator is the proxy duty to set the correct channel id.
You could  have a global id generator that we use to create these channels id, we should call this id generator from `open_standard_channel_down` and assign an unique id to the channel.